File: server/models/device.js
module.exports = (sequelize, DataTypes) => {
    const Device = sequelize.define('Device', {
        serialNumber: { type: DataTypes.STRING, primaryKey: true },

        modelName: { type: DataTypes.STRING, allowNull: false },
        modelIdentifier: { type: DataTypes.STRING, allowNull: false },
        modelNumber: { type: DataTypes.STRING, allowNull: false },

        hardwareUUID: { type: DataTypes.STRING, allowNull: false },

        createdAt: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
        updatedAt: { type: DataTypes.DATE, defaultValue: DataTypes.NOW }
    }, {
        tableName: 'devices'
    })

    return Device
};

File: app-iOS/Inventory Manager/Models/Device.swift
//
//  Device.swift
//  InventoryManager
//
//  Created by Ryan Mackin on 12/11/20.
//

import Foundation

struct Device: Codable {
    var serialNumber: String
    var modelName: String
    var modelIdentifier: String
    var modelNumber: String
    var hardwareUUID: String
}
